src/lib/vm/vm.js

```diff
@@ -621,6 +622,16 @@ class VmStack {
       );
     }
 
+    if (element === "GlobalStateProvider") {
+      if(!attributes.value) {
+        throw new Error("GlobalStateProvider requires a value prop");
+      }
+      if(!isArray(attributes.value) || attributes.value.length !== 2) {
+        throw new Error("Invalid value prop for GlobalStateProvider");
+      }
+      this.vm.globalStateContext = attributes.value;
```

Comment:
1. I don't understand this part. Shouldn't we just create a wrapper of React.Context?

2. Why inject [state, setState] here instead of context by name?